src/Integration.Vsix/SonarLintTagger/IssuesSnapshot.cs

```diff
@@ -121,11 +119,21 @@ public override bool TryGetValue(int index, string columnName, out object conten
 
         private string GetHelpLink(string ruleKey)
         {
-            // ruleKey is in format "javascript:1234"
-            // ruleId is "1234"
-            string ruleId = ruleKey.Substring(ruleKey.IndexOf(':') + 1);
+            var colonIndex = ruleKey.IndexOf(':');
+            // ruleKey is in format "javascript:S1234" (or javascript:SOMETHING for legacy keys)
+
             // language is "javascript"
-            string language = ruleKey.Substring(0, ruleKey.IndexOf(':'));
+            var language = ruleKey.Substring(0, colonIndex);
+
+            // ruleId should be "1234" (or SOMETHING for legacy keys)
+            var ruleId = ruleKey.Substring(colonIndex + 1);
+            if (ruleId.Length >= 2 &&
+                ruleId[0] == 'S' &&
+                char.IsDigit(ruleId[1]))
```

Comment:
I'd check all remaning digits. This probably doesn't matter, but it is a bit confusing why only the first character after S is checked.
`ruleId.Skip(1).All(char.IsDigit)`

Reply:
Because no legacy key contains numbers so there is no need to check for more.